bindings/node/test/clientEncryption.test.js

```diff
@@ -330,4 +368,276 @@ describe('ClientEncryption', function () {
+        'latin1'
+      );
+      let rndPos = 0;
+      sandbox.stub(cryptoCallbacks, 'randomHook').callsFake((buffer, count) => {
+        if (rndPos + count > rndData) {
+          return new Error('Out of fake random data');
+        }
+        buffer.set(rndData.subarray(rndPos, rndPos + count));
+        rndPos += count;
+        return count;
+      });
+
+      // stubbed out for AWS unit testing below
+      sandbox.stub(StateMachine.prototype, 'fetchKeys').callsFake((client, ns, filter, cb) => {
+        filter = deserialize(filter);
+        const keyIds = filter.$or[0]._id.$in.map(key => key.toString('hex'));
+        const fileNames = keyIds.map(
+          keyId => `${__dirname}/../../../test/data/keys/${keyId.toUpperCase()}-local-document.json`
+        );
+        const contents = fileNames.map(filename => EJSON.parse(fs.readFileSync(filename)));
+        cb(null, contents);
+      });
+    });
+
+    // This exactly matches _test_encrypt_fle2_explicit from the C tests
+    it('should explicitly encrypt and decrypt with the "local" KMS provider (FLE2, exact result)', function () {
+      const encryption = new ClientEncryption(new MockClient(), {
+        keyVaultNamespace: 'client.encryption',
+        kmsProviders: { local: { key: Buffer.alloc(96) } }
+      });
+
+      const encryptOptions = {
+        keyId: new Binary(Buffer.from('ABCDEFAB123498761234123456789012', 'hex'), 4),
+        indexKeyId: new Binary(Buffer.from('12345678123498761234123456789012', 'hex'), 4),
+        algorithm: 'Unindexed'
+      };
+
+      return encryption
+        .encrypt('value123', encryptOptions)
+        .then(encrypted => {
+          expect(encrypted._bsontype).to.equal('Binary');
+          expect(encrypted.sub_type).to.equal(6);
+          expect(encrypted.toString('base64')).to.equal(
+            'BqvN76sSNJh2EjQSNFZ4kBICTQaVZPWgXp41I7mPV1rLFTtw1tXzjcdSEyxpKKqujlko5TeizkB9hHQ009dVY1+fgIiDcefh+eQrm3CkhQ=='
+          );
+
+          return encryption.decrypt(encrypted);
+        })
+        .then(decrypted => {
+          expect(decrypted).to.equal('value123');
+        });
+    });
+  });
+
+  context('FLE2 explicit encryption spec tests', function () {
+    // cf. https://github.com/mongodb/specifications/blob/078ed5ff4d9216cbb906bec0002693c90878afd1/source/client-side-encryption/tests/README.rst#explicit-encryption
+
+    const ENCRYPTED_FIELDS = EJSON.parse(
+      fs.readFileSync(`${__dirname}/data/encryptedFields.json`),
+      { relaxed: false }
+    );
+    const KEY1_DOCUMENT = EJSON.parse(fs.readFileSync(`${__dirname}/data/key1-document.json`), {
+      relaxed: false
+    });
+    const KEY1_ID = KEY1_DOCUMENT._id;
+    const LOCAL_MASTERKEY =
+      'Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk';
+
+    let keyVaultClient;
+    let encryptedClient;
+    let clientEncryption;
+
+    async function setup() {
```

Comment:
The spec states that the explicit encryption tests must not run against a standalone server. Could we create a ticket to address that separately as there's a good bit of setup we need to do to make that happen.

Reply:
Sure – https://jira.mongodb.org/browse/NODE-4236 :+1:

Reply:
Thanks!